smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -286,4 +292,96 @@ static void writeHostPrefix(GenerationContext context, OperationShape operation)
             });
         });
     }
+
+    /**
+     * Writes a function used to dispatch event to corresponding event deserializers.
+     * This function assumes a event-specific deserialization function is generated
+     * for each returned structures.
+     *
+     * @param context The generation context.
+     * @param events The union of of events bond to an event stream
+     * @return A list of all event structure shapes for the event stream that were dispatched to.
+     */
+    static Set<StructureShape> generateDeserializingEventUnion(
```

Comment:
It's good that these are package private for now, but do they need to exist on this class since they're only used in the `HttpBindingProtocolGenerator`?